### PR TITLE
pending payment page functionality

### PR DESCRIPTION
**Pull request by dev-shrivastav-74**

### CODE REVIEW

File: src/store/pendingPayments/index.js [hunk 1]
```diff
@@ -0,0 +1,229 @@
+import { mayaClient } from '@/services/api';
+import { getPaymentAppTypeLabel } from '@/constant/enums';
+
+const extractPendingPayments = (response) => {
```
Comment (arrogant-ant) [inline src/store/pendingPayments/index.js]: why we need this. Maya response should have consistent contract.

File: tests/store/pendingPayments/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+            expect(store.state.pendingPayments.hasError).toBe(true);
+            expect(store.state.pendingPayments.errorMessage).toBe(
+                'Update failed ( Validation )',
+            );
+            expect(store.state.pendingPayments.pendingPayments).toEqual([
+                { PaymentId: 77, BookingId: 'BK-77' },
+            ]);
+        });
+    });
+});
```
Comment (arrogant-ant) [inline tests/store/pendingPayments/index.spec.js:213]: include edge cases like empty/negative amounts and missing UPI details

File: src/store/pendingPayments/index.js [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        paymentDetails =
+            paymentAppLabel !== '' ? `${upiId}/${paymentAppLabel}` : upiId;
+    } else if (mobile !== '') {
+        paymentDetails =
+            paymentAppLabel !== '' ? `${mobile}/${paymentAppLabel}` : mobile;
+    }
+
+    return {
+        ...payment,
+        PaymentId: Number(payment.PaymentId || payment.PaymentID || 0),
```
Comment (arrogant-ant) [inline src/store/pendingPayments/index.js]: why both naming conventions is needed to be handled

File: src/store/pendingPayments/index.js [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        account.PaymentApp ??
+            account.paymentApp ??
+            payment.PaymentApp ??
+            payment.paymentApp ??
+            0,
+    );
+    const paymentAppLabel =
+        pickAccountValue(account, ['PaymentAppLabel', 'paymentAppLabel']) ||
+        resolvePaymentAppLabel(paymentAppValue);
+    const upiId = pickAccountValue(account, ['UpiId', 'UpiID'], {
```
Comment (arrogant-ant) [inline src/store/pendingPayments/index.js]: here too

File: src/store/pendingPayments/index.js [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+    if (stripSpaces) {
+        normalizedValue = normalizedValue.replace(/\s+/g, '');
+    }
+    if (uppercase) {
+        normalizedValue = normalizedValue.toUpperCase();
+    }
+    return normalizedValue;
+};
+
+const pickAccountValue = (account, keys, options = {}) => {
```
Comment (arrogant-ant) [inline src/store/pendingPayments/index.js]: why are we getting multiple keys

File: src/views/PendingPaymentsPortal.vue [hunk 6]
```diff
@@ -0,0 +1,1015 @@
...
+            ).trim();
+            if (accountFullName !== '') {
+                return accountFullName;
+            }
+            return String(this.selectedPayment?.SoName || '').trim();
+        },
+        resolvedPaymentAppLabel() {
+            return String(this.accountDetails.PaymentAppLabel || '').trim();
+        },
+        resolvedAccountInfo() {
```
Comment (arrogant-ant) [inline src/views/PendingPaymentsPortal.vue:444]: extract this as a utility method and use from spot details page too.

**Review by arrogant-ant — CHANGES_REQUESTED** [review]

File: src/router/routes.js [hunk 7]
```diff
@@ -147,6 +148,11 @@ export const routes = [
         name: 'booking-portal',
         component: () => import('@/views/BookingPortal.vue'),
     },
+    {
```
Comment (arrogant-ant) [inline src/router/routes.js:187]: we can also add a route gaurd here

```
{
    path: pages.PENDING_PAYMENTS,
    name: 'pending-payments',
    component: () => import('@/views/PendingPaymentsPortal.vue'),
    beforeEnter: (to, from, next) => {
        const userState = store.state.user;
        if (userState.isAuthReady && userState.isAdmin) {
            next();
        } else if (!userState.isAuthReady) {
            // auth still loading -- could watch for it or redirect to login
            next({ name: 'login', query: { redirect: to.fullPath } });
        } else {
            next({ name: 'home' });
        }
    },
},
```

**Review by arrogant-ant — CHANGES_REQUESTED** [review]

File: src/router/routes.js (hunk 7 above)
Comment (dev-shrivastav-74) [reply src/router/routes.js:187]: done

File: src/store/pendingPayments/index.js (hunk 5 above)
Comment (dev-shrivastav-74) [reply src/store/pendingPayments/index.js]: ok sir

File: src/views/PendingPaymentsPortal.vue (hunk 6 above)
Comment (dev-shrivastav-74) [reply src/views/PendingPaymentsPortal.vue:444]: done sir

File: src/store/pendingPayments/index.js (hunk 4 above)
Comment (dev-shrivastav-74) [reply src/store/pendingPayments/index.js]: done

File: src/store/pendingPayments/index.js (hunk 3 above)
Comment (dev-shrivastav-74) [reply src/store/pendingPayments/index.js]: done

File: tests/store/pendingPayments/index.spec.js (hunk 2 above)
Comment (dev-shrivastav-74) [reply tests/store/pendingPayments/index.spec.js:213]: done

File: src/store/pendingPayments/index.js (hunk 1 above)
Comment (dev-shrivastav-74) [reply src/store/pendingPayments/index.js]: done

File: src/store/pendingPayments/index.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    async getPendingPayments({ commit }) {
+        commit('set-loading', true);
+
+        try {
+            const res = await mayaClient.get('/internal/pending-payments');
+
+            if (res?.DisplayMsg) {
+                commit(
+                    'set-error',
+                    res.DisplayMsg + ' ( ' + res.ErrorMsg + ' )',
```
Comment (arrogant-ant) [inline src/store/pendingPayments/index.js]: If ErrorMsg is missing, users see "undefined" in the message.

Fix: Coalesce, e.g. res.ErrorMsg ?? '', or template literals with guards.

File: src/router/routes.js [hunk 9]
```diff
@@ -147,6 +149,26 @@ export const routes = [
...
+        path: pages.PENDING_PAYMENTS,
+        name: 'pending-payments',
+        component: () => import('@/views/PendingPaymentsPortal.vue'),
+        beforeEnter: (to, from, next) => {
+            const userState = store.state.user;
+
+            if (!userState.isAuthReady) {
```
Comment (arrogant-ant) [inline src/router/routes.js]: Wait until isAuthReady is true (or use a promise/callback from your auth bootstrap), then decide next() vs redirect; or

**Review by arrogant-ant — CHANGES_REQUESTED** [review]

File: src/store/pendingPayments/index.js (hunk 8 above)
Comment (dev-shrivastav-74) [reply src/store/pendingPayments/index.js]: done

File: src/router/routes.js (hunk 9 above)
Comment (dev-shrivastav-74) [reply src/router/routes.js]: done

File: src/router/routes.js [hunk 10]
```diff
@@ -1,6 +1,27 @@
 import Home from '../views/PageHome.vue';
 import PageAbout from '@/views/PageAbout.vue';
 import { APP_LINK } from '../constant/constant';
+import store from '@/store';
+
+const waitForAuthReady = () => {
```
Comment (arrogant-ant) [inline src/router/routes.js]: add timeout / failure path so navigation cannot hang.

Comment (arrogant-ant) [inline src/router/routes.js]: add timeout / failure path so navigation cannot hang.

**Review by arrogant-ant — CHANGES_REQUESTED** [review]

File: src/router/routes.js (hunk 10 above)
Comment (dev-shrivastav-74) [reply src/router/routes.js]: done

Comment (dev-shrivastav-74) [reply src/router/routes.js]: done

File: src/store/pendingPayments/index.js [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+        } finally {
+            commit('set-loading', false);
+        }
+    },
+
+    async updateAmountToSO({ commit }, payload) {
+        commit('set-loading', true);
+
+        const reqBody = {
+            PaymentID: Number(payload?.PaymentID || 0),
```
Comment (arrogant-ant) [inline src/store/pendingPayments/index.js]: don't pass default value. IF the value is not present throw error.

**Review by arrogant-ant — CHANGES_REQUESTED** [review]

File: src/store/pendingPayments/index.js (hunk 11 above)
Comment (dev-shrivastav-74) [reply src/store/pendingPayments/index.js]: done sir

File: src/store/pendingPayments/index.js [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+
+    async updateAmountToSO({ commit }, payload) {
+        commit('set-loading', true);
+
+        try {
+            const paymentID = Number(payload?.PaymentID);
+            if (
+                payload?.PaymentID === undefined ||
+                payload?.PaymentID === null ||
+                Number.isNaN(paymentID)
```
Comment (arrogant-ant) [inline src/store/pendingPayments/index.js:69]: create a util method

**Review by arrogant-ant — APPROVED** [review]

File: src/views/PendingPaymentsPortal.vue [hunk 13]
```diff
@@ -642,10 +642,17 @@ export default {
         async recordPaymentSuccess() {
             if (!this.selectedPayment) return;
 
+            const paymentApp = this.selectedPayment?.Account?.PaymentApp;
```
Comment (arrogant-ant) [inline src/views/PendingPaymentsPortal.vue:645]: not needed

**Review by arrogant-ant — APPROVED** [review]

File: src/utils/isInvalidPaymentId.js [hunk 14]
```diff
@@ -0,0 +1,3 @@
+export const isInvalidPaymentId = (paymentID) => {
```
Comment (arrogant-ant) [inline src/utils/isInvalidPaymentId.js]: why is this method coupled to paymentID. whu not just check if it valid number?

Comment (arrogant-ant) [reply src/utils/isInvalidPaymentId.js]: not needed

File: tests/utils/isValidNumber.spec.js [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+        });
+
+        it('returns true for finite numbers', () => {
+            expect(isValidNumber(42)).toBe(true);
+        });
+
+        it('returns true for string values that are not NaN', () => {
+            expect(isValidNumber('42')).toBe(true);
+            expect(isValidNumber('')).toBe(true);
+            expect(isValidNumber('abc')).toBe(true);
```
Comment (arrogant-ant) [inline tests/utils/isValidNumber.spec.js]: how can these be valid numbers?

File: src/utils/isValidNumber.js [hunk 16]
```diff
@@ -0,0 +1,7 @@
+export const isValidNumber = (paymentID) => {
```
Comment (arrogant-ant) [inline src/utils/isValidNumber.js]: rename the variable name from paymentID to something generic

File: src/utils/isValidNumber.js [hunk 17]
```diff
@@ -0,0 +1,7 @@
...
+    return (
+        paymentID !== undefined &&
+        paymentID !== null &&
+        !Number.isNaN(paymentID)
```
Comment (arrogant-ant) [inline src/utils/isValidNumber.js]: export const isValidNumber = (value) => {
    if (value === null || value === undefined) return false;
    if (typeof value === 'string' && value.trim() === '') return false;
    const parsedValue = Number(value);
    return Number.isInteger(parsedValue);
};

**Review by arrogant-ant — CHANGES_REQUESTED** [review]

File: src/utils/isInvalidPaymentId.js (hunk 14 above)
Comment (dev-shrivastav-74) [reply src/utils/isInvalidPaymentId.js]: ok sir

File: tests/utils/isValidNumber.spec.js (hunk 15 above)
Comment (dev-shrivastav-74) [reply tests/utils/isValidNumber.spec.js]: by mistake

File: src/utils/isValidNumber.js (hunk 16 above)
Comment (dev-shrivastav-74) [reply src/utils/isValidNumber.js]: ok sir

File: src/utils/isValidNumber.js (hunk 17 above)
Comment (dev-shrivastav-74) [reply src/utils/isValidNumber.js]: ok sir

**Review by arrogant-ant — APPROVED** [review]
